### PR TITLE
fix: updates template condition for Arc

### PR DESCRIPTION
**Pull request by nilekhc**
Signed-off-by: Nilekh Chaudhari <1626598+nilekhc@users.noreply.github.com>

**Reason for Change**:

**Requirements**

- [x] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes: https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/972
**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [ ] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml [hunk 1]
```diff
@@ -22,13 +22,17 @@ spec:
...
         prometheus.io/scrape: "true" 
         prometheus.io/port: "8898"
 {{- end }}
+{{- end }}
 # this will ensure pods are rolled in next helm upgrade if cert files are updated.
-{{- if and .Values.enableArcExtension .Values.Azure.proxySettings.isProxyEnabled }}
+{{- if .Values.enableArcExtension }}
+{{- if .Values.Azure.proxySettings.isProxyEnabled }}
         checksum/arc-proxy-config: {{ include (print $.Template.BasePath "/arc-proxy-secret.yaml") . | sha256sum }}
+{{- end}}
```
Comment (sozercan) [inline charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml:35]:
```suggestion
{{- end }}
```

File: charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml [hunk 2]
```diff
@@ -22,13 +22,17 @@ spec:
 {{- if .Values.linux.podAnnotations}}
 {{- toYaml .Values.linux.podAnnotations | nindent 8 }}
 {{- end }}
-{{- if and .Values.enableArcExtension .Values.arc.enableMonitoring }}
```
Comment (sozercan) [inline charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml:25]: `and` didn't work here?

Comment (nilekhc) [reply charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml:25]: nope. It tries to evaluate `.Values.arc.enableMonitoring` as well which is not in values.yaml

**Review by aramase — COMMENTED** [review]
The `scan_images` failure should be resolved with https://github.com/Azure/secrets-store-csi-driver-provider-azure/pull/976.

Were you able to repro this? IIRC, in the last PR the order in which `and` operands were validated?

**Comment by nilekhc** [comment]
> The `scan_images` failure should be resolved with #976.
> 
> Were you able to repro this?

Yup.

**Comment by nilekhc** [comment]
> The `scan_images` failure should be resolved with #976.
> 
> Were you able to repro this? IIRC, in the last PR the order in which `and` operands were validated?

Yeah. I remember this. But I also remember when I first implemented this I had the same issue and exactly the same was the implementation to avoid nil pointer.

**Comment by aramase** [comment]
> > The `scan_images` failure should be resolved with #976.
> > Were you able to repro this? IIRC, in the last PR the order in which `and` operands were validated?
> 
> Yeah. I remember this. But I also remember when I first implemented this I had the same issue and exactly the same was the implementation to avoid nil pointer.

Is this a simple repro when not using an arc cluster? Is there a way we can catch this in CI/repro it with CI so we can validate this fix and ensure we don't hit this again?

**Comment by nilekhc** [comment]
> > > The `scan_images` failure should be resolved with #976.
> > > Were you able to repro this? IIRC, in the last PR the order in which `and` operands were validated?
> > 
> > 
> > Yeah. I remember this. But I also remember when I first implemented this I had the same issue and exactly the same was the implementation to avoid nil pointer.
> 
> Is this a simple repro when not using an arc cluster? Is there a way we can catch this in CI/repro it with CI so we can validate this fix and ensure we don't hit this again?

I have added a pod annotation in the e2e tests in the helm install. So if there is an issue, CI will give an error.

**Review by aramase — APPROVED** [review]

**Pull request merged**